modules/dns_common/dns_common.rb

```diff
@@ -24,5 +27,43 @@ def dns_find key
     rescue Resolv::ResolvError
       false
     end
+
+    def ptr_to_ip ptr
+     if ptr =~ /\.in-addr\.arpa$/
+       ptr.split('.')[0..-3].reverse.join('.')
+     elsif ptr =~ /\.ip6\.arpa$/
+       ptr.split('.')[0..-3].reverse.each_slice(4).inject([]) {|address, word| address << word.join}.join(":")
+     else
+       raise Proxy::Dns::Error.new("Not a PTR record: '#{ptr}'")
+     end
+    end
+
+    # conflict methods return values:
+    # no conflict: -1; conflict: 1, conflict but record / ip matches: 0
+    def a_record_conflicts(fqdn, ip)
+      if ip_addr = to_ipaddress(ip)
+        addresses = resolver.getaddresses(fqdn)
+        return -1 if addresses.empty?
+        return 0 if addresses.any? {|a| IPAddr.new(a) == ip_addr}
+        1
+      else
+        raise Proxy::Dns::Error.new("Not an IP Address: '#{ip}'")
+      end
+    end
+
+    def ptr_record_conflicts(fqdn, ip)
+      if ip_addr = to_ipaddress(ip)
+        names = resolver.getnames(ip_addr.to_s)
+        return -1 if names.empty?
+        return 0 if names.any? {|n| n.to_s.casecmp(fqdn) == 0}
```

Comment:
Was my first try. But here comes rubocop, the reason I was using #casecmp in the fist place:

```ruby
Offenses:

modules/dns_common/dns_common.rb:58:37: C: Use casecmp instead of downcase ==.
        return 0 if names.any? {|n| n.to_s.downcase == fqdn.to_s.downcase}
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```